ONSdigital/eq-author-app
# EAR-1707 - Grand summary of multiple calculated summaries

Signed-off-by: sudeep <sudeep.kunhikannan@ons.gov.uk>

### What is the context of this PR?

https://jira.ons.gov.uk/browse/EAR-1707

### How to review

**Please note that the 'Runner' does not support the 'Grand Summary' calculation yet, hence the 'Grand Summary' results will not be displayed while viewing the survey. Hence the feature can be verified by checking the convert schema output as explained below.**

- Create a new questionnaire - a section with two number answers, a calculated summary which adds the two number answers
- Create a second section similar to the above - two number answers and a calculated summary.
- Create a third section - add two percentage answers, then add a calculated summary page, label this as the 'Grand Summary' page. In the 'Answer Selector', all the answers created in the first and second sections must appear.
- That is, the number/percentage answers along with the calculated summary answers in the current/previous sections will be available for selection in the 'Grand Summary' page.
- Check if the labels, titles, error messages etc. are displayed as given in the Figma design prototype give in the ticket.
- Check if the 'Search' bar returns the expected results while entering a search text so that the answers are filtered.
- When a 'number' or 'percentage' answer is selected, the 'Calculated Summary' answers will be disabled and vice versa.
- Similarly, when a 'percentage' answer is selected, all other 'number' answer types will be disabled and vice versa.
- Select the two 'calculated summary' answers in the first and second sections to calculate the 'Grand Summary' page.
- In the '/convert' schema JSON output, check if the 'Grand Summary' page has the structure as given below with the 'block' ids of the calculated summary pages in the previous sections.

{
	"id": "blockc78b6f68-a571-45b4-b03c-efea3bd7dee1",
	"type": "CalculatedSummary",
	"title": "Grand Summary",
	"page_title": "Grand Summary",
	"calculation": {
		"operation": {
			"+": [{
					"identifier": "block170537aa-5d2f-4c6e-ad61-f7470fbb09cd",
					"source": "calculated_summary"
				},
				{
					"identifier": "block0d907a35-1a44-4cdc-9ed6-ebbc40eba6eb",
					"source": "calculated_summary"
				}
			]
		},
		"title": "Grand Summary"
	}
}


### What to do after everything is green

1. - [ ] Bring this branch up-to-date with Origin/Master
2. - [ ] If there are a lot of commits or some are not helpful to read, squash them down
3. - [ ] Click the **Merge** button on this pull request
4. - [ ] Does this change mean the **Capability examples** questionnaire should be updated?
5. - [ ] Move the Jira ticket for this task into the next stage of the process
